### ISSUE TITLE
Exception at reading 'uiMode' when clicked from settings on the home page

### ISSUE THREAD

**Issue by Taha-249** [jupyterlab/jupyterlab-desktop#857]
## Description
When there is no python environment found, jupyter labs cannot create new session. However, by changing on start up to create a new session generates this exception. There is the following blank screen.

![image](https://github.com/user-attachments/assets/9eda3308-db7c-4f3f-91e2-af3f86a3274e)

on clicking the hamburger button, it generates the following exception.

![image](https://github.com/user-attachments/assets/dc3cd9ed-dda1-4da1-a464-873e62d237cb)

Reinstalling program does not seem to fix the issue.

- Operating System and version: Windows 10 22H2 
- JupyterLab-Desktop version: latest

**Comment by JasonWeill** [comment]
@Taha-249 Thank you for opening this issue! Do you see any information in the logs when you start JupyterLab Desktop? You can find info about logs here: https://github.com/jupyterlab/jupyterlab-desktop/blob/master/troubleshoot.md#Logs

**Comment by JasonWeill** [comment]
@Taha-249 To follow up, can you please check for any info in the logs on startup and add it as a comment to this issue? Thanks!

**Comment by JasonWeill** [comment]
Without info from the logs, or any other reports, I'm going to close this issue. Please reopen or open a new issue if you continue to see problems in JupyterLab. Thanks!

**Comment by krassowski** [comment]
I encountered this on a fresh installation on Ubuntu 24.04 with `pyenv` installed. When I run `jlab . --no-sandbox` I saw no contents

![image](https://github.com/user-attachments/assets/d52bc576-b482-4178-8954-22e838472751)

Upon pressing the hamburger icon I saw:

![image](https://github.com/user-attachments/assets/4e992c29-ea5b-494d-ad53-d9811ea1c6cf)

The logs are:

```
[2024-09-20 11:08:03.055] [error] Python path "/home/krassowski/.config/jupyterlab-desktop/jlab_server/bin/python" does not exist.
[2024-09-20 11:08:05.243] [error] Failed to create free server Error: No default environment found!
    at /snap/jupyterlab-desktop/31/resources/app.asar/build/out/main/registry.js:282:32
[2024-10-01 12:45:16.909] [error] Python path "/home/krassowski/.config/jupyterlab-desktop/jlab_server/bin/python" does not exist.
[2024-10-01 12:45:19.902] [error] Failed to get environment info at path '/home/krassowski/.pyenv/shims/python'. Error: Command failed: /home/krassowski/.pyenv/shims/python -c import os, sys, json, platform, importlib

env_type = 'system'
env_name = 'python'
python_version = platform.python_version()
default_kernel = 'python3'

requirements = [
  'jupyterlab'
]

versions = {
  "python": python_version
}

for requirement in requirements:
  try:
    module = importlib.import_module(requirement)
    versions[requirement] = module.__version__
  except:
    versions[requirement] = 'NOT-FOUND'

if (getattr(sys, "base_prefix", None) or getattr(sys, "real_prefix", None) or sys.prefix) != sys.prefix:
  env_type = 'venv'

if env_type != 'venv' and os.path.exists(os.path.join(sys.prefix, "conda-meta")):
  is_root = os.path.exists(os.path.join(sys.prefix, "condabin"))
  env_type = 'conda-root' if is_root else 'conda-env'

if env_type != 'system':
  env_name = os.path.basename(sys.prefix)

try:
  import jupyter_client
  mkm = jupyter_client.multikernelmanager.MultiKernelManager()
  default_kernel = mkm.default_kernel_name
except:
  pass

print(json.dumps({"type" : env_type, "name": env_name, "versions" : versions, "defaultKernel": default_kernel}))

pyenv: python: command not found

The `python' command exists in these Python versions:
  3.9.11
  3.9.20

Note: See 'pyenv help global' for tips on allowing both
      python2 and python3 to be found.

    at ChildProcess.exithandler (node:child_process:430:12)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1091:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:514:28)
    at Pipe.<anonymous> (node:net:323:12)
[2024-10-01 12:45:20.402] [error] Failed to create free server Error: No default environment found!
    at /snap/jupyterlab-desktop/31/resources/app.asar/build/out/main/registry.js:282:32
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

I think that the robustness of the UI could be improved. It could also display a helpful message rather than crash.

**Comment by Nagidrop** [comment]
I can reproduce this error consistently: whenever I launch JupyterLab Desktop (normally) and quickly click on the hamburger menu icon (while window is still blank), the error dialog shows up. Then, the windows is still blank until you close the dialog. After closing, the app starts without further errors and functions normally.

I use custom Python environment (miniconda3) on latest version of JupyterLab (4.2.5) and JupyterLab Desktop (4.2.5-1) on Windows 11.

I have recorded a video for showcasing:

https://github.com/user-attachments/assets/e53e3c8d-21b5-46ed-aa85-46f76d1268eb

The logs show 2 warnings and 1 error but they appear only after JupyterLab Desktop has successfully launched. I attached them anyway:
```
[2024-10-04 20:04:14.770] [info]  In production mode
[2024-10-04 20:04:14.772] [info]  Logging to file (C:\Users\Nagi\AppData\Roaming\jupyterlab-desktop\logs\main.log) at 'info' level
[2024-10-04 20:04:24.355] [warn]  Unsatisfied version 2.3.2 from @jupyterlab/application-top of shared singleton module @lumino/widgets (required ^1.37.2)
[2024-10-04 20:04:25.256] [warn]  Language pack 'English_United States' not valid!
[2024-10-04 20:04:25.291] [verbose] Starting application in workspace: "default"
[2024-10-04 20:04:26.336] [error] Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'type')
[2024-10-04 20:04:26.615] [verbose] Starting WebSocket: ws://localhost:53679/api/kernels/a148d065-b663-42de-af19-4a3bd95d9c08
```

**Comment by notluquis** [comment]
Reproduced against master, and #831 is the same bug.

`ShowAppContextMenu` builds `uiModeSubmenu` eagerly at `sessionwindow.ts:915`, then spreads it conditionally 76 lines later:

```ts
...(inLocalSession ? uiModeSubmenu : []),
```

Because the array literal is built first, its four `checked:` expressions always evaluate, whether or not the submenu is used:

```ts
checked:
  this._contentViewType === ContentViewType.Lab &&
  this._labView.uiMode === UIMode.Zen
```

The `&&` guards on `_contentViewType`, not on `_labView`, and those two can disagree. `_labView` is declared `private _labView: LabView;` with no initializer (`:1834`), so it is `undefined` until `_loadLabView()` runs, while `_contentViewType` is set to `Lab` in the constructor. `_loadLabView` only runs inside `_createServerForSession().then(...)`, so any window that reaches `Lab` without a server sitting behind it hits an undefined read on the first hamburger click. That matches the "of undefined" in the message rather than the `null` that `_disposeSession` and `_updateContentView` assign.

Two ways in, both in the thread above:

| | |
| --- | --- |
| Permanent | the server never starts, `Failed to create free server Error: No default environment found!`, the `.catch` at `:225` shows the progress view and leaves the window at `Lab` forever |
| Transient | clicking the hamburger while the window is still blank, before the promise resolves, which is why dismissing the dialog and carrying on works |

Windows opened from the welcome view are safe: `_createSessionForRecent` awaits the server before setting `Lab`. The exposed starts are `StartupMode.NewLocalSession`, `LastSessions` with saved sessions, and a CLI-supplied session.

Verified by capturing the real `ShowAppContextMenu` handler in a throwaway test, putting the window in that state, and asserting the call throws `Cannot read properties of undefined (reading 'uiMode')`. Controls with a lab view present, and with the content view set to Welcome, do not throw. Mutation check: adding `?.` to the four reads changes the failure to a different error, so the assertion is pinned to this read rather than passing by accident.

Unchanged since the report. `git blame` puts the guard structure at four commits from February 2024, all before this issue, and `v4.2.1-1` has the same four reads.

A fix is one file. The narrow version is `this._labView?.uiMode`; the honest one is building `uiModeSubmenu` inside the `inLocalSession` branch, since the eager construction is the actual defect and it is only ever spread when that flag is true. Belongs under #1050, which already lists this and #831.

Worked through this with Claude Code; the reproduction and the mutation check were run here, the startup wiring was read.
